cwltool/sandboxjs.py

```diff
@@ -131,15 +123,39 @@ def new_js_proc(force_docker_pull=False, js_console=False):
     return nodejs
 
 
-def execjs(js, jslib, timeout=None, force_docker_pull=False, debug=False, js_console=False):  # type: (Union[Mapping, Text], Any, int, bool, bool, bool) -> JSON
+def exec_js_process(js_text, timeout=None, js_console=False, context=None, force_docker_pull=False, debug=False):
+    # type: (Text, int, bool, Text, bool, bool) -> Tuple[int, Text, Text]
+
+    if not hasattr(localdata, "procs"):
+        localdata.procs = {}
+
+    if js_console and context is not None:
+        raise NotImplementedError("js_console=True and context not implemented")
+
+    if js_console:
+        js_engine = 'cwlNodeEngineJSConsole.js'
+        _logger.warn("Running with support for javascript console in expressions (DO NOT USE IN PRODUCTION)")
+    elif context is not None:
+        js_engine = "cwlNodeEngineWithContext.js"
+    else:
+        js_engine = 'cwlNodeEngine.js'
 
-    if not hasattr(localdata, "proc") or localdata.proc.poll() is not None or onWindows():
-        localdata.proc = new_js_proc(force_docker_pull=force_docker_pull, js_console=js_console)
+    if localdata.procs.get(js_engine) is None \
+            or localdata.procs[js_engine].poll() is not None \
+            or onWindows():
+        res = resource_stream(__name__, js_engine)
+        js_engine_code = res.read().decode('utf-8')
 
-    nodejs = localdata.proc
+        localdata.procs[js_engine] = new_js_proc(js_engine_code, force_docker_pull=force_docker_pull)
```

Comment:
Seems to be causing neverending loops on windows: https://ci.appveyor.com/project/mr-c/cwltool/build/.773-master/job/o57lcfpp9xjcrc7i#L1775